### PR TITLE
Minor refactoring of the `Consistbl.Inconsistency` exception

### PR DESCRIPTION
**Pull request by gasche**
This is a minor refactoring of code in cmi_format and consistbl that I was trying to massage for a further change (not ready yet). The commits should be strictly behavior-preserving, but one exception changed its type.

### CODE REVIEW

File: typing/env.ml [hunk 1]
```diff
@@ -712,7 +712,11 @@ let check_consistency ps =
               add_import name;
               Consistbl.check crc_units name crc ps.ps_filename)
       ps.ps_crcs;
-  with Consistbl.Inconsistency(name, source, auth) ->
+  with Consistbl.Inconsistency {
+      unit_name = name;
+      inconsistent_source = source;
+      original_source = auth;
+    } ->
     error (Inconsistent_import(name, auth, source))
```
Comment (gasche) [inline typing/env.ml]: Note that here, the order of arguments is exchanged between `Inconsistency` and `Inconsistent_import`. This was already here in the original code. I don't know if the order of `Inconsistency_import` is intentional or a small mistake, but the semantics of the two arguments is the same (the error messages just says "foo is distinct from blah").

**Review by Armael — APPROVED** [review]
Looks good on principle, but CI's failing currently, it seems you need to update the code in some more places.

**Comment by gasche** [comment]
@Armael I rebased the code and fixed the places that were failing. Would you mind having a look again?

**Comment by Armael** [comment]
Should be good to go. Feel free to merge whenever.

**Comment by gasche** [comment]
(The CI failure is due to a missing `Changes`, but this doesn't need a Changes entry and the label is there.) Thanks! Merging.

**Pull request merged**